audit.sql

```diff
@@ -178,9 +182,39 @@ cannot obtain the active role because it is reset by the SECURITY DEFINER invoca
 of the audit trigger its self.
 $body$;
 
+CREATE OR REPLACE FUNCTION audit.create_table(audit_table_name text) RETURNS void AS $body$
+DECLARE
+BEGIN
+
+    EXECUTE 'CREATE TABLE IF NOT EXISTS audit.' || audit_table_name || '(
+    event_id bigserial primary key,
+    schema_name text not null,
+    table_name text not null,
+    relid oid not null,
+    session_user_name text,
+    action_tstamp_tx TIMESTAMP WITH TIME ZONE NOT NULL,
+    action_tstamp_stm TIMESTAMP WITH TIME ZONE NOT NULL,
+    action_tstamp_clk TIMESTAMP WITH TIME ZONE NOT NULL,
+    transaction_id bigint,
+    application_name text,
+    client_addr inet,
+    client_port integer,
+    client_query text,
+    action TEXT NOT NULL CHECK (action IN (''I'',''D'',''U'', ''T'')),
+    row_data jsonb,
+    changed_fields jsonb,
+    statement_only boolean not null
+    );';
+    EXECUTE 'REVOKE ALL ON audit.' || audit_table_name || ' FROM public;';
+    EXECUTE 'CREATE INDEX IF NOT EXISTS ' || audit_table_name || '_relid_idx ON audit.' || audit_table_name || '(relid);';
+    EXECUTE 'CREATE INDEX IF NOT EXISTS ' || audit_table_name || '_action_tstamp_tx_stm_idx ON audit.' || audit_table_name || '(action_tstamp_stm);';
+    EXECUTE 'CREATE INDEX IF NOT EXISTS ' || audit_table_name || '_action_idx ON audit.' || audit_table_name || '(action);';
+END;
+$body$
+language 'plpgsql';
 
 
-CREATE OR REPLACE FUNCTION audit.audit_table(target_table regclass, audit_rows boolean, audit_query_text boolean, ignored_cols text[]) RETURNS void AS $body$
+CREATE OR REPLACE FUNCTION audit.audit_table(target_table regclass, audit_rows boolean, audit_query_text boolean, ignored_cols text[], audit_table_name text) RETURNS void AS $body$
```

Comment:
What is your idea about changing the name of this function so that we may be able to use the old one just in case needed? @aminchini